### PR TITLE
Adding endpoints for crypto keys

### PR DESCRIPTION
**Pull request by Mikkel-Boje**
* GET: Keys, fetching a users keys; Fixes #42 
* POST: check keys, checking the servers keys vs. the users; Fixes #41

### CODE REVIEW

**Review by MJHC — DISMISSED** [review]
ok

File: client/src/key_management/bundle.rs [hunk 1]
```diff
@@ -84,22 +84,22 @@ impl KeyBundleContent {
...
-                Some((
-                    bundle.onetime_public_key_id.expect("").into(),
-                    PublicKey::deserialize(&(*bundle.onetime_public_key.to_owned().expect("")))
-                        .expect(""),
-                ))
-            };
+        let one_time_key: Option<(PreKeyId, PublicKey)> = if (bundle
+            .onetime_public_key_id
+            .is_none()
+            || bundle.onetime_public_key.is_none()
```
Comment (Darkros1245) [inline client/src/key_management/bundle.rs:90]: This should work, right?
```suggestion
            || bundle
            .onetime_public_key
            .is_none()
```

File: server/src/server.rs [hunk 2]
```diff
@@ -142,59 +121,77 @@ async fn redirect_http_to_https(addr: SocketAddr, http: u16, https: u16) -> Resu
     Ok(())
 }
 
+async fn handle_put_keys<S: SignalDatabase>(
+    database: S,
+    address: &ProtocolAddress,
+    bundle: DevicePreKeyBundle,
+) -> Result<()> {
+    todo!();
```
Comment (Darkros1245) [inline server/src/server.rs:129]: Is this still todo?

File: server/src/server.rs [hunk 3]
```diff
@@ -250,7 +247,7 @@ async fn put_registration_endpoint(State(state): State<SignalServerState<Postgre
 }
 
 /// Handler for the GET v2/keys endpoint.
-#[debug_handler]
+//#[debug_handler]
```
Comment (Darkros1245) [inline server/src/server.rs:250]: Is this no longer needed?

File: server/src/server.rs [hunk 4]
```diff
@@ -250,7 +247,7 @@ async fn put_registration_endpoint(State(state): State<SignalServerState<Postgre
...
 async fn get_keys_endpoint(State(state): State<SignalServerState<PostgresDatabase>>) {
     // TODO: Call `handle_get_keys`
```
Comment (Darkros1245) [inline server/src/server.rs:252]: Can't we do that now?

**Review by Darkros1245 — DISMISSED** [review]

**Review by Darkros1245 — APPROVED** [review]

**Review by UnluckyBird — DISMISSED** [review]

**Review by Darkros1245 — CHANGES_REQUESTED** [review]
NOT GUT

**Review by Darkros1245 — APPROVED** [review]

**Review by Laursen79 — APPROVED** [review]

**Pull request merged**